beetsPackages.alternatives: 0.13.3 -> 0.13.4

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
beets-minimal: fix tests

doronbehar committed:
beetsPackages.alternatives: 0.13.3 -> 0.13.4

Diff: https://github.com/geigerzaehler/beets-alternatives/compare/v0.13.3...v0.13.4

Changelog: https://github.com/geigerzaehler/beets-alternatives/blob/v0.13.4/CHANGELOG.md

azahi commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 445208`
Commit: `86a429ada70c954ab202d4e3a8013965cd7648f5`

---
### `x86_64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>beets-unstable (beetsPackages.beets-unstable)</li>
    <li>beets-unstable.dist (beetsPackages.beets-unstable.dist)</li>
    <li>beets-unstable.doc (beetsPackages.beets-unstable.doc)</li>
    <li>beets-unstable.man (beetsPackages.beets-unstable.man)</li>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
    <li>beetsPackages.filetote</li>
    <li>beetsPackages.filetote.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 21 packages built:</summary>
  <ul>
    <li>beets (beetsPackages.beets, beetsPackages.beets-stable, python313Packages.beets)</li>
    <li>beets.dist (beetsPackages.beets-stable.dist, beetsPackages.beets.dist, python313Packages.beets.dist)</li>
    <li>beets.doc (beetsPackages.beets-stable.doc, beetsPackages.beets.doc, python313Packages.beets.doc)</li>
    <li>beets.man (beetsPackages.beets-stable.man, beetsPackages.beets.man, python313Packages.beets.man)</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.audible</li>
    <li>beetsPackages.audible.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>python312Packages.beetcamp</li>
    <li>python312Packages.beetcamp.dist</li>
    <li>python312Packages.beets</li>
    <li>python312Packages.beets.dist</li>
    <li>python312Packages.beets.doc</li>
    <li>python312Packages.beets.man</li>
    <li>python313Packages.beetcamp</li>
    <li>python313Packages.beetcamp.dist</li>
    <li>supergee</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>beets-unstable</summary>
<pre>play								play
playlist							playlist
plexupdate							plexupdate
random								random
replace							      &lt;
replaygain							replaygain
rewrite								rewrite
scrub								scrub
smartplaylist							smartplaylist
sonosupdate							sonosupdate
spotify								spotify
subsonicplaylist						subsonicplaylist
subsonicupdate							subsonicupdate
substitute							substitute
the								the
thumbnails							thumbnails
types								types
unimported							unimported
web								web
zero								zero</pre>
</details>
<details>
<summary>beetsPackages.copyartifacts</summary>
<pre>    from beets.library import DefaultTemplateFunctions
E   ImportError: cannot import name &#x27;DefaultTemplateFunctions&#x27; from &#x27;beets.library&#x27; (/nix/store/7dgmyp3li454ps7jm3hly8m64m3gvwxq-beets-2.4.0/lib/python3.13/site-packages/beets/library/__init__.py)
=============================== warnings summary ===============================
../../nix/store/gnk1d1a4lk06r1qyacm38dam96jndh0z-python3.13-confuse-1.7.0/lib/python3.13/site-packages/confuse/util.py:126
  /nix/store/gnk1d1a4lk06r1qyacm38dam96jndh0z-python3.13-confuse-1.7.0/lib/python3.13/site-packages/confuse/util.py:126: DeprecationWarning: &#x27;pkgutil.get_loader&#x27; is deprecated and slated for removal in Python 3.14; use importlib.util.find_spec() instead
    loader = pkgutil.get_loader(name)

tests/helper.py:13
  /build/source/tests/helper.py:13: UserWarning: beets.mediafile is deprecated; use mediafile instead
    from beets import mediafile

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/test_filename.py
ERROR tests/test_flatdirectory.py
ERROR tests/test_nesteddirectory.py
ERROR tests/test_printignored.py
ERROR tests/test_reimport.py
!!!!!!!!!!!!!!!!!!! Interrupted: 5 errors during collection !!!!!!!!!!!!!!!!!!!!
======================== 2 warnings, 5 errors in 0.18s =========================</pre>
</details>
<details>
<summary>beetsPackages.filetote</summary>
<pre>=========================== short test summary info ============================
ERROR tests/test_audible_m4b_files.py
ERROR tests/test_exclude.py
ERROR tests/test_filename.py
ERROR tests/test_filesize_fixes.py
ERROR tests/test_flatdirectory.py
ERROR tests/test_manipulate_files.py
ERROR tests/test_music_files.py
ERROR tests/test_nesteddirectory.py
ERROR tests/test_pairing.py
ERROR tests/test_patterns.py
ERROR tests/test_printignored.py
ERROR tests/test_reimport.py
ERROR tests/test_rename.py
ERROR tests/test_rename_filetote_fields.py
ERROR tests/test_rename_inline_plugin.py
ERROR tests/test_rename_item_fields.py
ERROR tests/test_rename_paths.py
!!!!!!!!!!!!!!!!!!! Interrupted: 17 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 17 errors in 0.38s ==============================</pre>
</details>
</details>

doronbehar committed:
beets-unstable: alias to beets

To avoid the former left behind the latter.

doronbehar commented:
cc @aszlig for the beets-unstable aliasing.

returntoreality reviewed (approved):
looks good, copyartifacts and filetote need seperate fixes
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 445208`
Commit: `22012fafdb77a646af8639ba1d9a35c33f4c7cdc`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
    <li>beetsPackages.filetote</li>
    <li>beetsPackages.filetote.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 21 packages built:</summary>
  <ul>
    <li>beets (beetsPackages.beets, beetsPackages.beets-stable, python313Packages.beets)</li>
    <li>beets.dist (beetsPackages.beets-stable.dist, beetsPackages.beets.dist, python313Packages.beets.dist)</li>
    <li>beets.doc (beetsPackages.beets-stable.doc, beetsPackages.beets.doc, python313Packages.beets.doc)</li>
    <li>beets.man (beetsPackages.beets-stable.man, beetsPackages.beets.man, python313Packages.beets.man)</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.audible</li>
    <li>beetsPackages.audible.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>python312Packages.beetcamp</li>
    <li>python312Packages.beetcamp.dist</li>
    <li>python312Packages.beets</li>
    <li>python312Packages.beets.dist</li>
    <li>python312Packages.beets.doc</li>
    <li>python312Packages.beets.man</li>
    <li>python313Packages.beetcamp</li>
    <li>python313Packages.beetcamp.dist</li>
    <li>supergee</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>beetsPackages.copyartifacts</summary>
<pre>    from beets.library import DefaultTemplateFunctions
E   ImportError: cannot import name &#x27;DefaultTemplateFunctions&#x27; from &#x27;beets.library&#x27; (/nix/store/7dgmyp3li454ps7jm3hly8m64m3gvwxq-beets-2.4.0/lib/python3.13/site-packages/beets/library/__init__.py)
=============================== warnings summary ===============================
../../nix/store/gnk1d1a4lk06r1qyacm38dam96jndh0z-python3.13-confuse-1.7.0/lib/python3.13/site-packages/confuse/util.py:126
  /nix/store/gnk1d1a4lk06r1qyacm38dam96jndh0z-python3.13-confuse-1.7.0/lib/python3.13/site-packages/confuse/util.py:126: DeprecationWarning: &#x27;pkgutil.get_loader&#x27; is deprecated and slated for removal in Python 3.14; use importlib.util.find_spec() instead
    loader = pkgutil.get_loader(name)

tests/helper.py:13
  /build/source/tests/helper.py:13: UserWarning: beets.mediafile is deprecated; use mediafile instead
    from beets import mediafile

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/test_filename.py
ERROR tests/test_flatdirectory.py
ERROR tests/test_nesteddirectory.py
ERROR tests/test_printignored.py
ERROR tests/test_reimport.py
!!!!!!!!!!!!!!!!!!! Interrupted: 5 errors during collection !!!!!!!!!!!!!!!!!!!!
======================== 2 warnings, 5 errors in 0.17s =========================</pre>
</details>
<details>
<summary>beetsPackages.filetote</summary>
<pre>=========================== short test summary info ============================
ERROR tests/test_audible_m4b_files.py
ERROR tests/test_exclude.py
ERROR tests/test_filename.py
ERROR tests/test_filesize_fixes.py
ERROR tests/test_flatdirectory.py
ERROR tests/test_manipulate_files.py
ERROR tests/test_music_files.py
ERROR tests/test_nesteddirectory.py
ERROR tests/test_pairing.py
ERROR tests/test_patterns.py
ERROR tests/test_printignored.py
ERROR tests/test_reimport.py
ERROR tests/test_rename.py
ERROR tests/test_rename_filetote_fields.py
ERROR tests/test_rename_inline_plugin.py
ERROR tests/test_rename_item_fields.py
ERROR tests/test_rename_paths.py
!!!!!!!!!!!!!!!!!!! Interrupted: 17 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 17 errors in 0.35s ==============================</pre>
</details>
</details>

dschrempf commented:
My beets build is still failing. I have not followed this discussion, but let me know if there is anything I can do to speed up merging the fix! Thanks for working on this!

aszlig commented:
> My beets build is still failing.

If you mean the failures with `copyartifacts` and `filetote`... these also fail for the merge-base (bd9861062ce030e486c92822dca78f552fae41d5), so they should be unrelated to these changes.

doronbehar committed:
beetsPackages.copyartifacts: mark as broken

doronbehar committed:
beetsPackages.filetote: 1.0.2 -> 1.1.0

Diff: https://github.com/gtronset/beets-filetote/compare/v1.0.2...v1.1.0

Changelog: https://github.com/gtronset/beets-filetote/blob/v1.1.0/CHANGELOG.md

doronbehar committed:
beetsPackages.filetote: mark as broken

doronbehar commented:
@dschrempf I suspect you are experiencing an issue due to an overlay in which you specify plugins. This is worth investigating. Could you please give more details on how do you build your beets? With/Without this patch. I will merge it anyway now that CI is green.
